### ISSUE TITLE
Enhancement: [semi-spacing] spaces before semicolon in type definition is not detected

### ISSUE THREAD

**Issue by so1ve** [typescript-eslint/typescript-eslint#6681]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/semi-spacing

### Description

The space before the semicolon in type definition is not detected.

### Fail

```typescript
type Foo = 42 ;
```

### Pass

```typescript
type Foo = 42;
```

### Additional Info

I have made a small eslint plugin to cover this case: https://github.com/so1ve/eslint-config/blob/main/packages/eslint-plugin/src/rules/semi-spacing.ts. I will be glad to send a pr to fix this problem!

**Comment by so1ve** [comment]
sorry I didn't recognize that typescript-eslint doesn't wrap the original semi-spacing rule...
